app/controllers/shelters_controller.rb

```diff
@@ -14,8 +14,16 @@ def create
               state: params[:state],
               zip: params[:zip]
               })
-    shelter.save
-    redirect_to '/shelters'
+              
+    missing_fields = shelter_params.select{|_,v| v.nil? || v == ""}.keys
```

Comment:
I assume `_` refers to the unused hash key and `v` to value? Let me know if that's wrong. But if that's an accurate understanding, we should be more descriptive with the block variable naming of the value, so that our code is more readable. Maybe `user_input` or `params_value` depending on what range of params we have here?

Reply:
That said, this is an awesome setup for figuring out which form fields are missing

Reply:
Yeah I normally do k,v for key/value and use a _ to show I am not using the key in the loop

Reply:
Can change the v to value?